Essentials/src/com/earth2me/essentials/MetaItemStack.java

```diff
@@ -477,57 +472,53 @@ public Enchantment getEnchantment(final User user, final String name) throws Exc
-        if (MaterialUtil.isBanner(stack.getType()) && !stack.getType().toString().equals("SHIELD") && string != null) {
+        if (MaterialUtil.isBanner(stack.getType()) && string != null) {
             final String[] split = splitPattern.split(string, 2);
 
             if (split.length < 2) {
-                throw new Exception(tl("invalidBanner", split[1]));
+                throw new Exception(tl("invalidBanner", string));
             }
 
-            PatternType patternType = null;
-            try {
-                patternType = PatternType.valueOf(split[0]);
-            } catch (Exception ignored ) {}
+            DyeColor baseColor = null;
+            org.bukkit.block.banner.Pattern pattern = null;
 
-            final BannerMeta meta = (BannerMeta) stack.getItemMeta();
             if (split[0].equalsIgnoreCase("basecolor")) {
-                Color color = Color.fromRGB(Integer.valueOf(split[1]));
-                meta.setBaseColor(DyeColor.getByColor(color));
-            } else if (patternType != null) {
-                PatternType type = PatternType.valueOf(split[0]);
-                DyeColor color = DyeColor.getByColor(Color.fromRGB(Integer.valueOf(split[1])));
-                org.bukkit.block.banner.Pattern pattern = new org.bukkit.block.banner.Pattern(color, type);
-                meta.addPattern(pattern);
+                baseColor = ColorUtil.getDyeColor(split[1]);
+            } else {
+                try {
+                    PatternType type = PatternType.valueOf(split[0].toUpperCase());
+                    DyeColor color = ColorUtil.getDyeColor(split[1]);
+                    pattern = new org.bukkit.block.banner.Pattern(color, type);
+                } catch (IllegalArgumentException ignored) {
+                }
             }
 
-            stack.setItemMeta(meta);
-        } else if (stack.getType().toString().equals("SHIELD") && string != null) {
-            final String[] split = splitPattern.split(string, 2);
-
-            if (split.length < 2) {
-                throw new Exception(tl("invalidBanner", split[1]));
-            }
+            if (stack.getType().toString().equals("SHIELD")) {
+                // Hacky fix for accessing Shield meta - https://github.com/drtshock/Essentials/pull/745#issuecomment-234843795
+                BlockStateMeta meta = (BlockStateMeta) stack.getItemMeta();
+                Banner banner = (Banner) meta.getBlockState();
 
-            PatternType patternType = null;
-            try {
-                patternType = PatternType.valueOf(split[0]);
-            } catch (Exception ignored ) {}
+                if (baseColor != null) {
+                    banner.setBaseColor(baseColor);
+                } else if (pattern != null) {
+                    if (VersionUtil.getServerBukkitVersion().isHigherThanOrEqualTo(VersionUtil.BukkitVersion.fromString("1.13.0-R0.1-SNAPSHOT"))) {
```

Comment:
I know this is temporary but you should use the existing VersionUtil constants, or add a new one if missing.

Reply:
Not sure why I didn't do that in the first place 🤦‍♂